src/plugins/intel_gpu/src/graph/primitive_inst.cpp

```diff
@@ -508,7 +508,9 @@ event::ptr primitive_inst::realloc_if_needed() {
     }
 
     // Clear out memory if if was previously reused, but now primitive can't be optimized
-    if (_node->is_type<gather>() && !can_be_optimized() && _outputs[0] && _network.get_engine().is_the_same_buffer(dep_memory(0), output_memory(0))) {
+    if (_node->is_type<gather>() && !can_be_optimized() && _outputs[0]
+        && (_impl_params->input_layouts[0].count() != 0 // check if layout is zero dimension before using dep_memory()
+            && _network.get_engine().is_the_same_buffer(dep_memory(0), output_memory(0)))) {
```

Comment:
Will this case happen? 
In the below condition, if input_layouts[0] is empty, it is marked as can_be_optimized here. 
So !can_be_optimized() && empty input shouldnt happen at the same time.

Reply:
In some mask rcnn model, output is not empty but input is empty tensor with nullptr. 
To prevent null pointer exception, I add check whether memory pointer is null or not. 

`calc_output_layouts: gather:Gather_13108 input tensor: f16:bfyx:0:nopad => empty data(memory is nullptr)`
`calc_output_layouts: gather:Gather_13108 input tensor: i32:bfyx:1:nopad`
`calc_output_layouts: gather:Gather_13108 output tensor: f16:bfyx:1:nopad`

Reply:
please review it again. thanks.